### PR TITLE
Bugfix/flush buffers on call hold

### PR DESCRIPTION
**Pull request by cspiel1**
This is a rework on pull request https://github.com/alfredh/baresip/pull/707 .

### CODE REVIEW

File: src/stream.c [hunk 1]
```diff
@@ -258,6 +258,11 @@ static void rtp_handler(const struct sa *src, const struct rtp_header *hdr,
 			     sdp_media_name(s->sdp), s->ssrc_rx, hdr->ssrc,
 			     mbuf_get_left(mb), src);
 		}
+		else if (hdr->m) {
+			flush = true;
+			info("stream: %s: marker bit set\n",
+				sdp_media_name(s->sdp));
+		}
```
Comment (alfredh) [inline src/stream.c]: the marker bit has different semantics for audio/video stream.

audio: signal start of a new talk spurt
video: signal a complete video frame

I think checking RTP header marker bit should be done in audio.c instead.

Comment (cspiel1) [reply src/stream.c]: Right! Now it looks better.

**Comment by alfredh** [comment]
@cspiel1 thanks for helping out with these improvements :)

**Comment by cspiel1** [comment]
you're welcome!
r,chris

**Pull request merged**